common/types/activation.go

```diff
@@ -291,6 +292,11 @@ func (atx *ActivationTx) CalcAndSetID() error {
 	if atx.Signature == nil {
 		return fmt.Errorf("cannot calculate ATX ID: sig is nil")
 	}
+
+	if atx.MsgHash != BytesToHash(atx.HashInnerBytes()) {
+		return fmt.Errorf("bad message hash")
+	}
```

Comment:
no it doesn't. 

SetMetadata is only called when ATX needs to be signed. i.e. it should only be called in the ATX building flow (see `SignAndFinalizeAtx`).

for gossiped/synced ATXs, Metadata should be already set correctly. otherwise it is syntactically invalid. hence the check here. Metadata is the actual data that gets signed.